### PR TITLE
roachtest: use non root authentication by default

### LINKED ISSUE #63145: roachtest: make secure/non-root the default for new tests
Most of our roachtests use the root user on insecure clusters, which is not realistic. Now that we have [one roachtest](https://github.com/cockroachdb/cockroach/blob/af0e2cb6f347413ea370591524a3e1dbe49464b4/pkg/cmd/roachtest/connection_latency.go#L27) which does the proper thing, let's look at making that the default behavior for future tests. This should also make it easier to switch over existing tests.

Jira issue: CRDB-6444

### PR DESCRIPTION
**Pull request by DarrylWong**
This PR attempts to minimize the usage of root user authentication in roachtests, as the root user skips certain authentication paths. 

This is done through changing the default mode of authentication to `AuthUserCert` instead of `AuthRootCert`. This establishes non root user auth as the preferred default for roachtests and forces tests to explicitly opt into root auth. Relevant roachtest/roachprod helpers such as `c.Conn` and `{pgurl}` will now default to using the `DefaultUser` instead of root.

The majority of tests should now be authenticating with a non root user. The exceptions are:
1. Tests that run in insecure mode. 
3. `multitenant` tests that use the old API. The old API does not create a default admin user for the tenant or copy certs to the tenant. While the `c2c` tests have helpers that do this, it would be easier/better to just switch the problematic tests to the new API than to try and reuse the c2c API. I did this for a few of the trivial switches, i.e `kv0/enc=false/nodes=3/cpu=32/mt-shared-process`.
4. Roachprod cluster setup that cannot use a non root user, i.e. the command to create the default non root user.

Release note: none
Epic: none
Fixes: https://github.com/cockroachdb/cockroach/issues/63145

### CODE REVIEW

**Comment by DarrylWong** [comment]
[Ran the GCE suite here](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14125018?buildTab=overview&hideProblemsFromDependencies=false&expandBuildTestsSection=true&hideTestsFromDependencies=false&expandBuildChangesSection=true). Only a few related test failures that have been fixed. Rerunning those + preempted tests [here](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14125018?buildTab=overview&hideProblemsFromDependencies=false&expandBuildTestsSection=true&hideTestsFromDependencies=false&expandBuildChangesSection=true)

File: pkg/cmd/roachtest/cluster.go [hunk 1]
```diff
@@ -1349,19 +1349,28 @@ func (c *clusterImpl) FetchDebugZip(
...
-			// Ignore the files in the the log directory; we pull the logs separately anyway
+			// Ignore the files in the log directory; we pull the logs separately anyway
 			// so this would only cause duplication.
 			excludeFiles := "*.log,*.txt,*.pprof"
+
 			cmd := roachtestutil.NewCommand("%s debug zip", test.DefaultCockroachPath).
 				Option("include-range-info").
 				Flag("exclude-files", fmt.Sprintf("'%s'", excludeFiles)).
-				Flag("url", fmt.Sprintf("{pgurl:%d}", node)).
+				Flag("url", pgurl[0]).
```
Comment (srosenberg) [inline pkg/cmd/roachtest/cluster.go:1368]: Maybe rename to `nodePGUrl`? It wasn't immediately clear that `pgurl` is already node-specific. The diff probably misled me since before we were using a macro.

**Review by srosenberg — APPROVED** [review]
Nice!

File: pkg/cmd/roachtest/tests/import.go [hunk 2]
```diff
@@ -355,7 +355,7 @@ func registerImportDecommissioned(r registry.Registry) {
 		// Decommission a node.
 		nodeToDecommission := 2
 		t.Status(fmt.Sprintf("decommissioning node %d", nodeToDecommission))
-		c.Run(ctx, option.WithNodes(c.Node(nodeToDecommission)), `./cockroach node decommission --self --wait=all --url={pgurl:2}`)
+		c.Run(ctx, option.WithNodes(c.Node(nodeToDecommission)), fmt.Sprintf(`./cockroach node decommission --self --wait=all --port={pgport:%d} --certs-dir=%s`, nodeToDecommission, install.CockroachNodeCertsDir))
```
Comment (renatolabs) [inline pkg/cmd/roachtest/tests/import.go:358]: Why was this change needed?

File: pkg/cmd/roachtest/tests/kv.go [hunk 3]
```diff
@@ -118,7 +118,8 @@ func registerKV(r registry.Registry) {
 			}
 		}
 		if opts.sharedProcessMT {
-			createInMemoryTenant(ctx, t, c, appTenantName, c.Range(1, nodes), true /* secure */)
+			startOpts = option.DefaultStartSharedVirtualClusterOpts(appTenantName)
+			c.StartServiceForVirtualCluster(ctx, t.L(), c.Range(1, nodes), startOpts, install.MakeClusterSettings(), c.Range(1, nodes))
```
Comment (renatolabs) [inline pkg/cmd/roachtest/tests/kv.go:122]: Sort of proving your point in https://github.com/cockroachdb/cockroach/pull/119679#discussion_r1504730005: the third argument could be _anything_ and it's irrelevant in this case, since we're creating an in-memory tenant. 😓

**Review by renatolabs — APPROVED** [review]
Awesome! 🎉

File: pkg/cmd/roachtest/tests/import.go (hunk 2 above)
Comment (DarrylWong) [reply pkg/cmd/roachtest/tests/import.go:358]: `cockroach node` doesn't support non root authentication since it's an RPC connection. All the other `cockroach node` calls already used this `--port` and `--certs-dir` pattern, this was just the outlier.

File: pkg/cmd/roachtest/tests/kv.go (hunk 3 above)
Comment (DarrylWong) [reply pkg/cmd/roachtest/tests/kv.go:122]: Yeah I wasn't sure to leave it empty or have it be the same as the storage nodes. Both seemed mildly confusing to me 😅

**Comment by DarrylWong** [comment]
TFTRs!

bors r=srosenberg, renatolabs
